src/ParamHandler.ts

```diff
@@ -104,9 +106,19 @@ export class ParamHandler {
     private parseValue(value: any, paramMetadata: ParamMetadata) {
```

Comment:
we'll need to refactor this block later or, because its very hard to understand what is going on there